### DISCUSSION TITLE
self-hosted not using bashrc

### DISCUSSION THREAD

**Discussion by alexhraber** [community/community#25407] [Actions]
<p>I can source it, but by default nothing from bashrc is loaded into the runtime at launch.</p>
<p>Should I update a different file instead bashrc? We have several additions to our base image to support multiple engineering workflows and dependencies, including a rather lengthy PATH.</p>
<p>I’ve suggested people source bashrc, but typically this is something that is expected in the linux world that when you have a shell, the environment loads with any configurations.</p>
<p>Is there any plan to automatically source bashrc? </p>

**Comment by Yanjingzhu** [comment]
<p>Hi <a class="mention" href="/everops-alex">@everops-alex</a> , </p>
<p>Currently Github Actions invoke bash scripts in non-interactive mode, it will not load .bashrc by default.  </p>
<p><em>&gt;&gt; Is there any plan to automatically source bashrc? </em></p>
<p>I would recommend you submiting this requirement in the <a href="https://support.github.com/contact/feedback?contact%5Bcategory%5D=actions" rel="nofollow noopener">Feedback form for GitHub Actions</a>. </p>

**Comment by Jackenmen** [comment, marked as answer]
<p>You can use custom shell, <a class="mention" href="/everops-alex">@everops-alex</a>:</p>
<p><a href="https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#custom-shell" rel="nofollow noopener">https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#custom-shell</a></p>
<p>For example, the template of “bash” shell is:</p>
<pre><code>bash --noprofile --norc -eo pipefail {0}
</code></pre>
<p>So to make it source the rc files before running the script, you can use (-l flag is needed to invoke bash as login shell, see “Invocation” section in bash’s man page for more info):</p>
<pre><code>bash -leo pipefail {0}
</code></pre>
<p>_Edited: _I used -I instead of -l in the reply</p>

**Reply by AllanOricil** [reply]
Does not work on my self-hosted runner

<img width="356" alt="image" src="https://user-images.githubusercontent.com/55927613/231462211-c35531fd-f58b-477f-a02b-5e09987e6006.png">


<img width="673" alt="image" src="https://user-images.githubusercontent.com/55927613/231462118-7b440048-b148-4808-9d90-7742ddf92fce.png">


this is my bash version
<img width="549" alt="image" src="https://user-images.githubusercontent.com/55927613/231463559-adc96319-b50c-43d5-b899-3b4d388c3950.png">

**Reply by FaureAlexis** [reply]
Hello @AllanOricil,
There is a typo in your shell property : It's `-leo` not `-ieo` (-l for login)

**Reply by snakajima-cogility** [reply]
Heads up this prevents early exit with `exit 0`.
![image](https://github.com/user-attachments/assets/d8928d10-782a-47b1-9c2b-3318132a5c98)

**Comment by xavierLowmiller** [comment]
<p>Is it possible to set this for the whole workflow?</p>

**Reply by carlos2martinize** [reply]
in all in all i dont think it was posible untill ray trace route followed a farensic file recording

**Comment by Yanjingzhu** [comment]
<p>You could use <a href="https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#defaultsrun" rel="nofollow noopener"> <strong><code>defaults.run</code></strong> </a>  to  provide default  <code>shell</code>  options for all <strong>run</strong> steps in a workflow.</p>
<pre><code class="lang-auto">name: use bashrc
defaults:
  run:
    shell: bash -ieo pipefail {0}
on: 
 push:
jobs: 
  job1:
    ......
</code></pre>

**Reply by devzer01** [reply]
the shell is declared as enum you can only type bash cant provide arguments

**Reply by devzer01** [reply]
ah my bad i just read the error message again ```Invalid shell option. Shell must be a valid built-in (bash, sh, cmd, powershell, pwsh) or a format string containing '{0}'```
